### DISCUSSION TITLE
Question about choice of tornado webserver - realtime kernel

### DISCUSSION THREAD

**Discussion by guysoft** [mod-audio/mod-ui#144] [Q&A]
Hey,
So I am playing with mod-ui on a raspberrypi with a realtime kernel. Its awesome, but sometimes the mod-ui interface just hangs. It might take like 10 minutes to load. I suspect its because of tornado, I didn't really test because I am not sure how to replace it (yet).
I can see images downloading in the speed that is reminiscent of dialup internet (from top to bottom, slowly).

I was wondering if you have an explanation why to use a single thread per process web browser?

**Comment by riban-bw** [comment]
I am seeing this behaviour on a Raspberry Pi 4. The browser loads the patch but overlays it with a spinning loading animation for about 10 minutes. 

I see this message in the log for browsepy:
```
/usr/local/lib/python3.7/dist-packages/unicategories_tools/cache.py:42: UserWarning: Incompatible unicategories database. Please reinstall unicategories module to regenerate it.
```
I wonder if this is relevant. I ran `pip3 install --force-reinstall unicategories` but this did not help.

![image](https://user-images.githubusercontent.com/3158323/189041471-d0ba52c7-bab4-4efc-bef4-504fd9924be6.png)

**Comment by riban-bw** [comment]
There seems to be something hanging the client. On faster / different architecture clients the page loads faster but on some clients it can take 10 minutes.
